### ISSUE TITLE
JAX: Deep Learning - train a brainwave model

### ISSUE THREAD

**Issue by 3C-SCSU** [3C-SCSU/Avatar#319]
Using the provided data, obtain the success rates for NAO control model with the brain. Use JAX Deep Learning.

Check on Wiki - adjust to JAX accordingly:

<img width="284" height="62" alt="Image" src="https://github.com/user-attachments/assets/c6b30699-3dde-4d02-9f48-fd16f2e6f548" />

Do a pull request, adding the new JAX code implementation and trained model in the directory:

<img width="307" height="25" alt="Image" src="https://github.com/user-attachments/assets/bca8e985-3ba4-449f-8c6a-d897bf8a4114" />

**VERY IMPORTANT:**
Yes, JAX can be used for deep learning without relying on scikit-learn.
JAX is a high-performance numerical computing library with automatic differentiation and XLA compilation, making it suitable for deep learning. However, it is not a full-fledged deep learning framework like PyTorch or TensorFlow, and it does not offer the high-level API and built-in functionalities for model building and training that scikit-learn provides for traditional machine learning tasks.
To perform deep learning with JAX, you would typically use it in conjunction with libraries specifically designed for building neural networks within the JAX ecosystem, such as:
[Flax: ](https://www.google.com/search?rlz=1C1GCEA_en&cs=0&sca_esv=09f2b9d0e2ee4b15&q=Flax&sa=X&ved=2ahUKEwj4kbfSp5WQAxVVw_ACHWzvFJwQxccNegQIDhAD&mstk=AUtExfBL6LoobjTV0eiQkhyAyNeq2icOdlNYm8Nm7oNxiTrtlaRWfUBjAmYTNd40HQOklI1x29oUFM0pX6QZBAzMOcwQmwjKvIK1guYdouCVv31O7ufqBpCiBID2a-VfnzECbJbDNUlNpq7pkbSFLDq1ITXw__75cgZQN88oXp2pYEnM4Ms&csui=3)A neural network library that provides a set of tools for defining, training, and evaluating neural networks in JAX.
[Haiku: ](https://www.google.com/search?rlz=1C1GCEA_en&cs=0&sca_esv=09f2b9d0e2ee4b15&q=Haiku&sa=X&ved=2ahUKEwj4kbfSp5WQAxVVw_ACHWzvFJwQxccNegQIERAD&mstk=AUtExfBL6LoobjTV0eiQkhyAyNeq2icOdlNYm8Nm7oNxiTrtlaRWfUBjAmYTNd40HQOklI1x29oUFM0pX6QZBAzMOcwQmwjKvIK1guYdouCVv31O7ufqBpCiBID2a-VfnzECbJbDNUlNpq7pkbSFLDq1ITXw__75cgZQN88oXp2pYEnM4Ms&csui=3)Another neural network library for JAX, offering a functional approach to defining models.
[Optax: ](https://www.google.com/search?rlz=1C1GCEA_en&cs=0&sca_esv=09f2b9d0e2ee4b15&q=Optax&sa=X&ved=2ahUKEwj4kbfSp5WQAxVVw_ACHWzvFJwQxccNegQIFRAD&mstk=AUtExfBL6LoobjTV0eiQkhyAyNeq2icOdlNYm8Nm7oNxiTrtlaRWfUBjAmYTNd40HQOklI1x29oUFM0pX6QZBAzMOcwQmwjKvIK1guYdouCVv31O7ufqBpCiBID2a-VfnzECbJbDNUlNpq7pkbSFLDq1ITXw__75cgZQN88oXp2pYEnM4Ms&csui=3)A library for gradient processing and optimization, providing various optimizers commonly used in deep learning.
These libraries leverage JAX's core capabilities (like automatic differentiation and JIT compilation) to enable efficient deep learning model development and training, without requiring scikit-learn.

_Reference:_
**Jax Deep Learning**
https://medium.com/@nimritakoul01/jax-a-beginners-tutorial-ca09b25a3f56

**Comment by 3C-SCSU** [comment]
Ticket solved: https://github.com/3C-SCSU/Avatar/pull/416
